Makefile

```diff
@@ -1,22 +1,29 @@
-.PHONY: pre build release image clean
+# See: https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63
+# For a list of valid GOOS and GOARCH values
+# Note: these can be overriden on the command line e.g. `make PLATFORM=<platform> ARCH=<arch>`
+PLATFORM=$(shell go env GOOS)
+ARCH=$(shell go env GOARCH)
 
+ifeq ("$(PLATFORM)", "windows")
+bin=compose-unpacker.exe
+else
+bin=compose-unpacker
+endif
 dist := dist
-bin := $(shell basename $(CURDIR))
-image := portainer/compose-unpacker
+image := portainer/compose-unpacker:latest
+.PHONY: pre build release image clean download-binaries
 
-pre:
-	mkdir -pv $(dist) 
+download-binaries:
+	@./setup.sh $(PLATFORM) $(ARCH)
 
-build: pre
-	GOOS="$(shell go env GOOS)" GOARCH="$(shell go env GOARCH)" CGO_ENABLED=0 go build --installsuffix cgo --ldflags '-s' -o $(bin)
-	mv $(bin) $(dist)/
+pre:
+	mkdir -pv $(dist)
```

Comment:
Suggest you move this line above the @./setup line in download-binaries.  It's kindof redundant